[13.0][IMP] base_ubl: add customer as an argument in _ubl_add_item method

FerranCforgeFlow opened the pull request:
- Add _customer_ as an argument in the __ubl_add_item_ method the same way as the seller, as default set to false
- Add also _customer_ as an argument in the __ubl_get_customer_product_code_ so the product code can be taken either from the customer or from the product
- Fixed: the current version of the _ubl_add_supplier_party and _ubl_add_customer_party was not allowing to add the cbc:CustomerAssignedID and the cbc:SupplierAssignedID respectively, if the partner for which we want to build the party block was our company.The if not company condition has been removed, and a call to the _ubl_get_customer_assigned_id() method is first added, so if it does not return False, the UBL field is added.

FerranCforgeFlow committed:
[13.0][FIX] base_ubl: add argument in _ubl_add_item and fixed _ubl_add_supplier_party and _ubl_add_customer_party
Add customer as an argument in the _ubl_add_item method the same way as the seller, as default set to false
Add also customer as an argument in the ubl_get_customer_product_code so the product code can be taken either from the customer or from the product
- Fixed: the current version of the _ubl_add_supplier_party and _ubl_add_customer_party was not allowing to add the cbc:CustomerAssignedID and the cbc:SupplierAssignedID respectively, if the partner for which we want to build the party block was our company. The 'if not company' condition has been removed, and a call to the _ubl_get_customer_assigned_id() method is first added, so if it does not return False, the UBL field is added.

FerranCforgeFlow commented:
@bealdav could you check?

bealdav commented:
Hi thanks @FerranCforgeFlow , it seems there are already merged code here ?
like in https://github.com/OCA/edi/pull/421/files

Maybe a rebase could only let real updates. Or I missed something ? Thanks

FerranCforgeFlow commented:
In this PR I am fixing the supplier and customer party logic which is not in the #421. However, I am adding one more argument to the _ubl_get_customer_product_code method I added in the #421, so I am not adding the whole method again. I am also changing the location of  the _ubl_get_customer_assigned_id method that I added in #421 since in my view it makes more sense to be defined before two methods that use it instead of between them.

bealdav commented:
OK I'll dive into. Thanks a lot

FerranCforgeFlow commented:
What's your view @bealdav ?

bealdav commented:
/ocabot merge minor
